internal/services/elasticsan/elastic_san_volume_snapshot_data_source.go

```diff
@@ -0,0 +1,147 @@
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validate.ElasticSanSnapshotName,
+		},
+
+		"volume_group_id": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+	}
+}
+
+func (r ElasticSANVolumeSnapshotDataSource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"creation_source": {
+			Type:     pluginsdk.TypeList,
+			Computed: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"source_id": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+				},
+			},
+		},
+		"source_volume_size_gib": {
+			Computed: true,
+			Type:     pluginsdk.TypeInt,
+		},
+		"volume_name": {
+			Computed: true,
+			Type:     pluginsdk.TypeString,
+		},
+	}
+}
+
+func (r ElasticSANVolumeSnapshotDataSource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ElasticSan.Snapshots
+
+			var state ElasticSANVolumeSnapshotDataSourceModel
+			if err := metadata.Decode(&state); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			volumeGroupId, err := snapshots.ParseVolumeGroupID(state.VolumeGroupId)
+			if err != nil {
+				return err
+			}
+
+			id := snapshots.NewSnapshotID(volumeGroupId.SubscriptionId, volumeGroupId.ResourceGroupName, volumeGroupId.ElasticSanName, volumeGroupId.VolumeGroupName, state.Name)
+
+			resp, err := client.VolumeSnapshotsGet(ctx, id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return fmt.Errorf("%s does not exist", id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+
+			state.VolumeGroupId = volumeGroupId.ID()
+			state.Name = id.SnapshotName
+			if model := resp.Model; model != nil {
+				// these properties are not pointer so we don't need to check for nil
+				state.SourceVolumeSizeGiB = pointer.From(model.Properties.SourceVolumeSizeGiB)
+				state.VolumeName = pointer.From(model.Properties.VolumeName)
+
+				createSource, err := FlattenElasticSANVolumeSnapshotCreateSource(model.Properties.CreationData)
+				if err != nil {
+					return err
+				}
+				state.CreateSource = *createSource
+			}
+			metadata.SetID(id)
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func FlattenElasticSANVolumeSnapshotCreateSource(input snapshots.SnapshotCreationData) (*[]ElasticSANVolumeSnapshotCreateSource, error) {
+	if input.SourceId == "" {
+		return &[]ElasticSANVolumeSnapshotCreateSource{}, nil
+	}
+
+	// for now source ID can only be ElasticSAN Volume ID
+	var sourceId string
+	parsedSourceId, err := volumes.ParseVolumeIDInsensitively(input.SourceId)
+	if err != nil {
+		return nil, fmt.Errorf("parsing source ID as ElasticSAN Volume ID: %+v", err)
+	}
+	sourceId = parsedSourceId.ID()
+
+	return &[]ElasticSANVolumeSnapshotCreateSource{
+		{
+			SourceId: sourceId,
+		},
+	}, nil
```

Comment:
Since only ElasticSAN Volumes are supported for now can we simplify this for the time being
```suggestion
	volumeId, err := volumes.ParseVolumeIDInsensitively(input.SourceId)
	if err != nil {
		return nil, err
	}

	return &[]ElasticSANVolumeSnapshotCreateSource{
		{
			SourceId: volumeId.ID(),
		},
	}, nil
```

Reply:
changed